sdk/keyvault/keyvault-admin/review/keyvault-admin.api.md

```diff
@@ -0,0 +1,101 @@
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import * as coreHttp from '@azure/core-http';
+import { PagedAsyncIterableIterator } from '@azure/core-paging';
+import { TokenCredential } from '@azure/core-http';
+
+// @public
+export interface AccessControlClientOptions extends coreHttp.PipelineOptions {
+    serviceVersion?: SUPPORTED_API_VERSIONS;
+}
+
+// @public
+export interface CreateRoleAssignmentOptions extends coreHttp.OperationOptions {
+}
+
+// @public
+export interface DeleteRoleAssignmentOptions extends coreHttp.OperationOptions {
+}
+
+// @public
+export interface GetRoleAssignmentOptions extends coreHttp.OperationOptions {
+}
+
+// @public
+export class KeyVaultAccessControlClient {
+    constructor(vaultUrl: string, credential: TokenCredential, pipelineOptions?: AccessControlClientOptions);
+    createRoleAssignment(roleScope: RoleAssignmentScope, name: string, roleDefinitionId: string, principalId: string, options?: CreateRoleAssignmentOptions): Promise<KeyVaultRoleAssignment>;
+    deleteRoleAssignment(roleScope: RoleAssignmentScope, name: string, options?: DeleteRoleAssignmentOptions): Promise<KeyVaultRoleAssignment>;
+    getRoleAssignment(roleScope: RoleAssignmentScope, name: string, options?: GetRoleAssignmentOptions): Promise<KeyVaultRoleAssignment>;
+    listRoleAssignments(roleScope: RoleAssignmentScope, options?: ListRoleAssignmentsOptions): PagedAsyncIterableIterator<KeyVaultRoleAssignment>;
+    listRoleDefinitions(roleScope: RoleAssignmentScope, options?: ListRoleDefinitionsOptions): PagedAsyncIterableIterator<KeyVaultRoleDefinition>;
+    readonly vaultUrl: string;
+}
+
+// @public
+export interface KeyVaultPermission {
+    actions?: string[];
+    dataActions?: string[];
+    notActions?: string[];
+    notDataActions?: string[];
+}
+
+// @public
+export interface KeyVaultRoleAssignment {
+    readonly id?: string;
+    readonly name?: string;
+    properties?: KeyVaultRoleAssignmentPropertiesWithScope;
+    readonly type?: string;
+}
+
+// @public
+export interface KeyVaultRoleAssignmentProperties {
+    principalId?: string;
+    roleDefinitionId?: string;
+}
+
+// @public
+export interface KeyVaultRoleAssignmentPropertiesWithScope extends KeyVaultRoleAssignmentProperties {
+    scope?: RoleAssignmentScope;
+}
+
+// @public
+export interface KeyVaultRoleDefinition {
+    assignableScopes?: string[];
+    description?: string;
+    readonly id?: string;
+    readonly name?: string;
+    permissions?: KeyVaultPermission[];
+    roleName?: string;
+    roleType?: string;
+    readonly type?: string;
+}
+
+// @public
+export const LATEST_API_VERSION = "7.1";
+
+// @public
+export interface ListRoleAssignmentsOptions extends coreHttp.OperationOptions {
+}
+
+// @public
+export interface ListRoleDefinitionsOptions extends coreHttp.OperationOptions {
+}
+
+// @public
+export type RoleAssignmentScope = "/" | "/keys" | string;
```

Comment:
this feels like a weird type, though I'm not sure what would be better

Reply:
The string part is because it accepts UUIDs. I wonder if we could have some UUID  validator as a type? That would be generally helpful in our clients.